sigma-schema.json

```diff
@@ -0,0 +1,246 @@
+        },
+        {
+          "const": "test",
+          "description": "The rule doesn't show any obvious false positives on a limited set of test systems"
+        },
+        {
+          "const": "experimental",
+          "description": "A new rule that hasn't been tested outside of lab environments and could lead to many false positives"
+        },
+        {
+          "const": "deprecated",
+          "description": "The rule is to replace or cover another one. The link between both rules is made via the `related` field"
+        },
+        {
+          "const": "unsupported",
+          "description": "The rule can not be used in its current state (special correlation log, home-made fields...etc.)"
+        }
+      ]
+    },
+    "description": {
+      "type": "string",
+      "description": "A short description of the rule and the malicious activity that can be detected",
+      "maxLength": 65535
+    },
+    "license": {
+      "type": "string",
+      "description": "License of the rule according the SPDX ID specification (https://spdx.dev/ids/)"
+    },
+    "author": {
+      "type": "string",
+      "description": "Creator of the rule. (can be a name, nickname, twitter handle, etc.)"
+    },
+    "references": {
+      "type": "array",
+      "description": "References to the source that the rule was derived from. These could be blog articles, technical papers, presentations or even tweets",
+      "items": {
+        "type": "string"
+      }
+    },
+    "date": {
+      "type": "string",
+      "description": "Creation date of the rule. Use the format YYYY/MM/DD",
+      "pattern": "^\\d{4}[/]\\d{2}[/]\\d{2}$"
+    },
+    "modified": {
+      "type": "string",
+      "description": "Last modification date of the rule. Use the format YYYY/MM/DD",
+      "pattern": "^\\d{4}[/]\\d{2}[/]\\d{2}$"
+    },
+    "logsource": {
+      "type": "object",
+      "description": "The log source that the rule is supposed to detect malicious activity in.",
+      "properties": {
+        "category": {
+          "type": "string",
+          "description": "Select all log files written by a certain group of products, like firewalls or web server logs"
+        },
+        "product": {
+          "type": "string",
+          "description": "Select all log outputs of a certain product, e.g. all Windows Eventlog types including \"Security\", \"System\", \"Application\" and the new log types like \"AppLocker\" and \"Windows Defender\""
+        },
+        "service": {
+          "type": "string",
+          "description": "Select only a subset of a product's logs, like the \"sshd\" on Linux or the \"Security\" Eventlog on Windows systems"
+        },
+        "definition": {
+          "type": "string",
+          "description": "Describe the log source, including some information on the log verbosity level or configurations that have to be applied"
```

Comment:
As the properties of the logsource are optional according to the spec. I suggest we make it as a such. So its just need to be an object of the form `property: string`

Reply:
Fair enough!